beam/src/main/scala/magnolify/beam/RowType.scala

```diff
@@ -0,0 +1,281 @@
+    new BeamSchemaField[T] {
+      override type FromT = logicaltypes.OneOfType.Value
+      override type ToT = logicaltypes.OneOfType.Value
+
+      private def enumName(sub: Subtype[Typeclass, T]): String =
+        s"${sub.typeName.owner}.${sub.typeName.short}"
+
+      @transient private lazy val beamOneOfTypeCache
+        : concurrent.Map[ju.UUID, logicaltypes.OneOfType] = concurrent.TrieMap.empty
+      private def beamOneOfType(cm: CaseMapper): logicaltypes.OneOfType =
+        beamOneOfTypeCache.getOrElseUpdate(
+          cm.uuid,
+          logicaltypes.OneOfType.create(
+            sealedTrait.subtypes.map { sub =>
+              Field.of(enumName(sub), sub.typeclass.fieldType(cm))
+            }.asJava
+          )
+        )
+
+      override def fieldType(cm: CaseMapper): FieldType =
+        FieldType.logicalType(beamOneOfType(cm))
+      def from(v: logicaltypes.OneOfType.Value)(cm: CaseMapper): T = {
+        val idx = v.getCaseType.getValue
+        sealedTrait.subtypes.find(_.index == idx) match {
+          case None      => throw new IllegalArgumentException(s"OneOf index not found: [$idx]")
+          case Some(sub) => sub.typeclass.fromAny(v.getValue)(cm)
+        }
+      }
+
+      def to(v: T)(cm: CaseMapper): logicaltypes.OneOfType.Value =
+        sealedTrait.split(v)(sub =>
+          beamOneOfType(cm).createValue(enumName(sub), sub.typeclass.to(sub.cast(v))(cm))
+        )
+    }
+
+  def join[T](caseClass: CaseClass[Typeclass, T]): BeamSchemaField[T] = {
+    if (caseClass.isValueClass) {
+      val p = caseClass.parameters.head
+      val tc = p.typeclass
+      new BeamSchemaField[T] {
+        override type FromT = tc.FromT
+        override type ToT = tc.ToT
+        override def fieldType(cm: CaseMapper): FieldType = tc.fieldType(cm)
+        override def from(v: FromT)(cm: CaseMapper): T = caseClass.construct(_ => tc.fromAny(v)(cm))
+        override def to(v: T)(cm: CaseMapper): ToT = tc.to(p.dereference(v))(cm)
+      }
+    } else {
+      new Record[T] {
+        override def fieldType(cm: CaseMapper): FieldType = FieldType.row(schema(cm))
+
+        override protected def buildSchema(cm: CaseMapper): Schema =
+          caseClass.parameters
+            .foldLeft(Schema.builder()) { case (s, p) =>
+              s.addField(cm.map(p.label), p.typeclass.fieldType(cm))
+            }
+            .build()
+
+        override def from(v: Row)(cm: CaseMapper): T =
+          caseClass.construct(p => p.typeclass.fromAny(v.getValue[Any](p.index))(cm))
+
+        override def to(v: T)(cm: CaseMapper): Row = {
+          val values = caseClass.parameters.map { p =>
+            p.typeclass.to(p.dereference(v))(cm).asInstanceOf[Object]
+          }
+          Row.withSchema(schema(cm)).addValues(values.asJava).build()
+        }
+      }
+    }
+  }
+
+  // BYTE	An 8-bit signed value
+  implicit val bsfByte: BeamSchemaField[Byte] = id[Byte](_ => FieldType.BYTE)
```

Comment:
rename all variables according to the new naming
```suggestion
  implicit val rfByte: BeamSchemaField[Byte] = id[Byte](_ => FieldType.BYTE)
```